src/main/java/gov/va/viers/cdi/emis/ws/endpoint/MilitaryInfoEndpoint.java

```diff
@@ -30,17 +42,130 @@
       localPart = "eMISmilitaryServiceEligibilityRequest")
   @ResponsePayload
   public JAXBElement<EMISmilitaryServiceEligibilityResponseType> getServiceEligibility(
-      @RequestPayload InputEdiPiOrIcn request) {
-    /*     This should probably be wrapped in some null checks, not sure what EMIS does in those cases
+      @RequestPayload InputEdiPiOrIcn request,
+      @SoapHeader(value = "{http://viers.va.gov/cdi/CDI/commonService/v2}inputHeaderInfo")
+          SoapHeaderElement soapHeaderElement) {
+
+    InputHeaderInfo requestSoapHeaders = new InputHeaderInfo();
+    requestSoapHeaders = getInputHeaderInfo(soapHeaderElement);
+
+    try {
+      if (!request.getEdipiORicn().getInputType().equals("EDIPI")) {
+        return generateMissingOrInvalidEdipiEssError(
+            requestSoapHeaders, "MIS-ERR-03", "INVALID_IDENTIFIER");
+      } else if (request.getEdipiORicn().getEdipiORicnValue().isEmpty()) {
+        return generateMissingOrInvalidEdipiEssError(
+            requestSoapHeaders, "MIS-ERR-02", "MISSING_EDIPI");
+      }
+    } catch (DatatypeConfigurationException e) {
+      LOGGER.error("error while generating essError", e);
+    }
+
+    /* This should probably be wrapped in some null checks, not sure what EMIS does in those cases
     though with regards to returning that there was bad input*/
     JAXBElement<gov.va.schema.emis.vdrdodadapter.v2.EMISmilitaryServiceEligibilityResponseType>
         dodResponse =
             dodClient.getMilitaryServiceEligibilityResponse(
                 request.getEdipiORicn().getEdipiORicnValue());
 
-    EMISmilitaryServiceEligibilityResponseType noJaxbResponse;
-    noJaxbResponse = EMISMapper.INSTANCE.mapEMISmilitaryServiceEligibilityResponseType(dodResponse.getValue());
+    try {
+      if (dodResponse.getValue().getESSError() != null) {
+        if ("ERROR".equals(dodResponse.getValue().getESSError().getESSResponseCode())
+            && "INVALID_EDIPI_INPUT".equals(dodResponse.getValue().getESSError().getText())) {
+          EMISMapperImpl emisMapper = new EMISMapperImpl();
+          ESSErrorType essErrorType =
+              emisMapper.mapESSErrorType(dodResponse.getValue().getESSError());
+          return generateBadFormatEdipiEssError(requestSoapHeaders);
+        }
+      }
+    } catch (DatatypeConfigurationException e) {
```

Comment:
If we're going to do these try/catches anyways, I'd rather we do it at the root of the issue (xmlGregorianCalendarCurrentTime) instead of wherever we want to make an error.